lib/PuppeteerSharp/Page.cs

```diff
@@ -312,6 +317,7 @@ public int DefaultNavigationTimeout
         public Accessibility Accessibility { get; }
 
         internal bool JavascriptEnabled { get; set; } = true;
+        internal bool HasPopupEventListeners => Popup?.GetInvocationList().Any() == true;
         #endregion
```

Comment:
Why do we need this?

Reply:
I can only access the Invocation list from the inside of the class.
So here https://github.com/kblok/puppeteer-sharp/pull/906/files#diff-304795bf53e0aba6fcc27d50cbc6d7dfR46 if  we have no listeners we don't need to await for `PageAsync`.